pkg/api/component_report.go

```diff
@@ -1183,6 +1184,9 @@ func getReportFromCacheOrGenerate[T any](c cache.Cache, cacheKey interface{}, ge
 		if err != nil {
 			return defaultVal, []error{err}
 		}
+		if bytes.Equal(cacheKey, []byte("{}")) {
+			return defaultVal, []error{fmt.Errorf("cache key serialized to an empty json struct, must have public fields")}
+		}
```

Comment:
I'm not sure this would get noticed, what do you think about panicking like this? https://github.com/openshift/sippy/pull/1333

Reply:
Updated to remove my version.